main.nf

```diff
@@ -488,12 +489,25 @@ if (!params.Q2imported){
 			script:
 			discard_untrimmed = params.retain_untrimmed ? '' : '--discard-untrimmed'
 			primers = single_end ? "--rc -g ${params.FW_primer}...${params.RV_primer}" : "-g ${params.FW_primer} -G ${params.RV_primer}"
+			int_out_files = single_end ? "-o firstcutadapt/${reads}" : "-o firstcutadapt/${reads[0]} -p firstcutadapt/${reads[1]}"
+			int_in_files = single_end ? "firstcutadapt/${reads}" : "firstcutadapt/${reads[0]} firstcutadapt/${reads[1]}"
 			out_files = single_end ? "-o trimmed/${reads}" : "-o trimmed/${reads[0]} -p trimmed/${reads[1]}"
 			in_files = single_end ? "${reads}" : "${reads[0]} ${reads[1]}"
 			"""
 			mkdir -p trimmed
-			cutadapt ${primers} ${discard_untrimmed} ${out_files} ${in_files} \
-				 > cutadapt_log_${pair_id}.txt
+			if [[ $params.double_primer=TRUE && $discard_untrimmed="--discard-untrimmed" ]]; then
+	                        mkdir -p firstcutadapt
+				cutadapt ${primers} ${discard_untrimmed} \
+					${int_out_files} \
+					${in_files} >> cutadapt_log_${pair_id}.txt
+                                cutadapt ${primers} --discard-trimmed \
```

Comment:
Why do you discard here all sequences that contain a second primer? Does that mean all (or many of) these sequences that contain a second primer are artifacts, i.e. not biological sequences, and should not be used further? Did you confirm this? 

A second occurrence of primers in the dataset means that the PCR went horribly wrong and I can imagine that sequences containing two primers are better discarded, I just want to be certain.

Reply:
So far we've only found limited cases with double primers, but in these cases it seems that the PCR did go horribly wrong and keeping these reads is detrimental to the analysis. We haven't been able to think of any cases where a second primer sequence would be of biological origin, but we made this step optional just in case someone can think of a biological reason for double primers.

Reply:
Alright, thanks, have never seen double primers.